common/netmap.go

```diff
@@ -0,0 +1,39 @@
+package common
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop"
+	"github.com/nspcc-dev/neo-go/pkg/interop/contract"
+	"github.com/nspcc-dev/neo-go/pkg/interop/convert"
+	"github.com/nspcc-dev/neo-go/pkg/interop/lib/address"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/management"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/std"
+	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
+)
+
+// RegisterNewEpochHook registers calling contract as a NewEpoch
+// callback requester. Netmap contract's address is taken from the
+// NNS contract, therefore, it must be presented and filled with
+// netmap information for a correct RegisterNewEpochHook call; otherwise
+// a successive call is not guaranteed.
+// Caller must have `NewEpoch` method with a single numeric argument.
+func RegisterNewEpochHook() {
+	nnsContract := management.GetContractByID(1)
+	if nnsContract == nil {
+		panic("missing NNS contract")
+	}
+
+	resolveRes := contract.Call(nnsContract.Hash, "resolve", contract.ReadOnly, "netmap.neofs", 16)
```

Comment:
Contract bindings would be helpful here, but OK for now.

Reply:
you mean neo-go has smth for contracts (not RPC) too? i know nothing about it

Reply:
you mean neo-go has smth for contracts (not RPC) too? i know nothing about it

Reply:
Try `./bin/neo-go contract generate-wrapper`.

Reply:
This will need to be done for all contracts, so for now we better leave it as is.